e2e/nodejs_image/foolib/BUILD.bazel

```diff
@@ -1,8 +1,14 @@
+load("@build_bazel_rules_nodejs//:index.bzl", "copy_to_bin")
 load("@build_bazel_rules_nodejs//internal/js_library:js_library.bzl", "js_library")
 
+copy_to_bin(
```

Comment:
FYI @gregmagolan our thought is that the copy feature can be orthogonal to js_library, and where composition is possible we should use it - thoughts?

Reply:
Since js_library is linkable with a LinkablePackageInfo I think the default behaviour should that everything is available via the link. Linker is linking to the output folder so it seems like a bad user experience for users to have to copy_to_bin all source files if they want them exposed.

Reply:
@mistic the `js_library` symbol we import everywhere is already a macro, so it's pretty easy to fold the copy_to_bin as a separate rule (keeping our separation of concerns) but defer this design change, which is kinda orthogonal.

Also makes this PR a lot smaller since most of the use sites won't have to change.
Then we could tackle this design question separately, WDYT?

Reply:
Maybe not that simple. Embedded I was filtering on `is_source` which is not available in a macro:
```
        if src.is_source and not src.path.startswith("external/"):
            dst = ctx.actions.declare_file(src.basename, sibling = src)
            if ctx.attr.is_windows:
                copy_cmd(ctx, src, dst)
            else:
                copy_bash(ctx, src, dst)
```

Reply:
copy_to_bin may barf if you pass it a file already in the output tree

Reply:
Perhaps an option to disable to auto-copy to bin and then the rule will need to decide if it is linking to the output tree or the source tree again if that is set? Seems like a power-user feature optimization feature to me.

Reply:
After reading the feedback both provided here, I believe I'll recover the old behaviour of auto-copy things into bin for now.

Reply:
Alright, done